### PR TITLE
Update timerbot example

### LINKED ISSUE #2144: [BUG] Update timerbot example
https://github.com/python-telegram-bot/python-telegram-bot/blob/3b4559dd95d14a4c9a9ca54237b5fe547386b666/examples/timerbot.py#L56-L58

If the job has already run, those lines give an error after v13.0, as APS remvos the job after it has run. But as jobs can't be persisted natively, I think we should just restructure the example to give the jobs the `chat_id` as name and use `get_jobs_by_name` to remove them if necessary

Edit: And we should definetely remove the `pass_*` kwargs, those are deprecated since v12 …

### PR DESCRIPTION
**Pull request by timqsh**
closes #2144

### CODE REVIEW

File: examples/timerbot.py [hunk 1]
```diff
@@ -42,6 +42,16 @@ def alarm(context):
...
 
 
+def remove_job_if_exists(name, context):
+    """Remove job with given name. Returns whether job was removed."""
+    current_jobs = context.job_queue.get_jobs_by_name(name)
+    job_removed = False
+    for job in current_jobs:
+        job.schedule_removal()
+        job_removed = True
+    return job_removed
```
Comment (Bibo-Joshi) [inline examples/timerbot.py:52]:
```suggestion
    current_jobs = context.job_queue.get_jobs_by_name(name)
    if not current_jobs:
        return False
    for job in current_jobs:
        job.schedule_removal()
    return True
```

File: examples/timerbot.py [hunk 2]
```diff
@@ -52,30 +62,24 @@ def set_timer(update, context):
...
-        return
-
-    job = context.chat_data['job']
-    job.schedule_removal()
-    del context.chat_data['job']
-
-    update.message.reply_text('Timer successfully unset!')
+    chat_id = update.message.chat_id
+    job_removed = remove_job_if_exists(str(chat_id), context)
+    text = 'Timer successfully unset!' if job_removed else 'You have no active timer'
```
Comment (Bibo-Joshi) [inline examples/timerbot.py:81]:
```suggestion
    text = 'Timer successfully cancelled!' if job_removed else 'You have no active timer.'
```

**Review by Bibo-Joshi — COMMENTED** [review]
Looks and works great :) I left two minor comments

**Comment by timqsh** [comment]
Done =)

**Comment by Bibo-Joshi** [comment]
Thanks for your contribution! :)

**Pull request merged**